### PR TITLE
protoc-gen-go-grpc: remove `use_generic_streams_experimental` flag (defaults to true)

### PR DESCRIPTION
**Pull request by eshitachandwani**
Fixes task 2 : #1894 - Delete the code to generate stream interfaces for client and server.

Deletes the unused code to make stream generics default.

RELEASE NOTES: None

### CODE REVIEW

**Comment by purnesh42H** [comment]
@eshitachandwani please try regenerating the protos and run the relevant examples to make sure everything is correct

File: cmd/protoc-gen-go-grpc/grpc.go [hunk 1]
```diff
@@ -333,11 +329,9 @@ func clientSignature(g *protogen.GeneratedFile, method *protogen.Method) string
 	if !method.Desc.IsStreamingClient() && !method.Desc.IsStreamingServer() {
 		s += "*" + g.QualifiedGoIdent(method.Output.GoIdent)
 	} else {
-		if *useGenericStreams {
-			s += clientStreamInterface(g, method)
-		} else {
-			s += method.Parent.GoName + "_" + method.GoName + "Client"
-		}
+
```
Comment (purnesh42H) [inline cmd/protoc-gen-go-grpc/grpc.go:332]: is 332 and 334 an  empty line?

**Comment by arvindbr8** [comment]
> @eshitachandwani please try regenerating the protos and run the relevant examples to make sure everything is correct

CI should already be doing that

File: cmd/protoc-gen-go-grpc/grpc.go [hunk 2]
```diff
@@ -460,11 +399,8 @@ func serverSignature(g *protogen.GeneratedFile, method *protogen.Method) string
 		reqArgs = append(reqArgs, "*"+g.QualifiedGoIdent(method.Input.GoIdent))
 	}
 	if method.Desc.IsStreamingClient() || method.Desc.IsStreamingServer() {
-		if *useGenericStreams {
-			reqArgs = append(reqArgs, serverStreamInterface(g, method))
-		} else {
-			reqArgs = append(reqArgs, method.Parent.GoName+"_"+method.GoName+"Server")
-		}
+		reqArgs = append(reqArgs, serverStreamInterface(g, method))
+
```
Comment (arvindbr8) [inline cmd/protoc-gen-go-grpc/grpc.go:403]: remove new line here pls

```suggestion
```

File: cmd/protoc-gen-go-grpc/grpc.go [hunk 3]
```diff
@@ -178,13 +178,9 @@ func generateFileContent(gen *protogen.Plugin, file *protogen.File, g *protogen.
...
-	if *useGenericStreams {
-		g.P("// Requires gRPC-Go v1.64.0 or later.")
-		g.P("const _ = ", grpcPackage.Ident("SupportPackageIsVersion9"))
-	} else {
-		g.P("// Requires gRPC-Go v1.62.0 or later.")
-		g.P("const _ = ", grpcPackage.Ident("SupportPackageIsVersion8")) // When changing, update version number above.
-	}
+	g.P("// Requires gRPC-Go v1.64.0 or later.")
+	g.P("const _ = ", grpcPackage.Ident("SupportPackageIsVersion9"))
+
```
Comment (arvindbr8) [inline cmd/protoc-gen-go-grpc/grpc.go:183]: remove new line pls. we should try to use newlines to create blocks of related code. 

```suggestion
```

**Review by arvindbr8 — APPROVED** [review]
changes LGTM, modulo two comments. 

Also instead of saying partially fixes #issue, we should say which part of the issue is fixed.  From the looks of it, it seems like the only thing pending is a release?

**Comment by eshitachandwani** [comment]
> Also instead of saying partially fixes #issue, we should say which part of the issue is fixed. From the looks of it, it seems like the only thing pending is a release?
Updating the gRPC.io to reflect the stream generics is also pending, but as far as grpc-go is concerned, yes only release is pending.

**Comment by eshitachandwani** [comment]
> @eshitachandwani please try regenerating the protos and run the relevant examples to make sure everything is correct
Done. It checks out correctly.

File: cmd/protoc-gen-go-grpc/grpc.go (hunk 1 above)
Comment (eshitachandwani) [reply cmd/protoc-gen-go-grpc/grpc.go]: Removed.

File: cmd/protoc-gen-go-grpc/grpc.go (hunk 2 above)
Comment (eshitachandwani) [reply cmd/protoc-gen-go-grpc/grpc.go]: Done

**Comment by purnesh42H** [comment]
> > @eshitachandwani please try regenerating the protos and run the relevant examples to make sure everything is correct
> 
> CI should already be doing that

that's great. good to know that

**Comment by arvindbr8** [comment]
relnotes here are used by the release bot for gRPC releases only.  relnotes for the plugin releases are handled manually.

**Pull request merged**